memif-attach/main.go

```diff
@@ -82,24 +141,49 @@ func createMemif(ctx context.Context, conn api.Connection, socketID uint32, mode
 	if err != nil {
 		log.Entry(ctx).Fatalln("ERROR: memifCreate failed:", err)
 	}
-	log.Entry(ctx).Infof("Memif interface created")
-	log.Entry(ctx).Infof("swIfIndex"+
-		"Role"+
-		"SocketID",
+	log.Entry(ctx).Infof("Memif interface created\n"+
+		"swIfIndex %v\n"+
+		"Role %v\n"+
+		"SocketID %v\n",
 		rsp.SwIfIndex, memifCreate.Role, memifCreate.SocketID)
-	// ifindex.Store(ctx, isClient, rsp.SwIfIndex)
+
+	// Set AdminUp
+	swinterface := &interfaces.SwInterfaceSetFlags {
+		SwIfIndex: rsp.SwIfIndex, 
+		Flags: interface_types.IF_STATUS_API_FLAG_ADMIN_UP,  
+	}
+	retVal, err := interfaces.NewServiceClient(conn).SwInterfaceSetFlags(ctx, swinterface)
+	if err != nil {
+		log.Entry(ctx).Fatalln("ERROR:  %v", err)
+	}
+	log.Entry(ctx).Infof("SwInterfaceSetFlags %v", retVal)
+
 	return nil
 }
 
-func dumpMemif(ctx context.Context, conn api.Connection) (memif.RPCService_MemifDumpClient, error) {
+func dumpMemif(ctx context.Context, conn api.Connection) () {
 	c := memif.NewServiceClient(conn)
 
-	MemifDumpClient, MemifDumpErr := c.MemifDump(ctx, &memif.MemifDump{})
-	if MemifDumpErr != nil {
-		log.Entry(ctx).Fatalln("ERROR: MemifDump failed:", MemifDumpErr)
+	memifDumpMsg, err := c.MemifDump(ctx, &memif.MemifDump{})
+	if err != nil {
+		log.Entry(ctx).Fatalln("ERROR: MemifDump failed:", err)
 	}
 	log.Entry(ctx).Infof("Socket file dump")
-	return MemifDumpClient, MemifDumpErr
+	// done := make(chan bool)
+	// for {
```

Comment:
So I tried the commented method to dump the message from Recv but got the above error, and I tried to figure out what I can get from the error message.

Reply:
This still need to be fixed. When call in a for loop, error raised.